src/OneMind.Persistence/Handlers/CreateOrganizationHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace OneMind.Handlers;
+
+sealed class CreateOrganizationHandler : IRequestHandler<CreateOrganizationRequest, int>
+{
+    readonly IDbContextFactory<OneMindDbContext> _contextFactory;
+    readonly InsertOrganization _insertOrganization;
+
+    public CreateOrganizationHandler(IDbContextFactory<OneMindDbContext> contextFactory, InsertOrganization insertOrganization)
+    {
+        _contextFactory = contextFactory;
+        _insertOrganization = insertOrganization;
+    }
+
+    public async Task<int> Handle(CreateOrganizationRequest request, CancellationToken cancellationToken)
+    {
+        using var context = await _contextFactory.CreateDbContextAsync(cancellationToken);
+        using var transaction = await context.Database.BeginTransactionAsync(cancellationToken);
+
+        var leader =  await context.Users.GetRequiredAsync(request.LeaderId, cancellationToken);
+
+        var id = await _insertOrganization.ExecuteAsync(
+            context.WithHotSave(),
+            new()
+            {
+                Name = request.Name,
+                Description = request.Description,
+                Leader = leader,
+                LeaderId = leader.Id
+            },
```

Comment:
// After LeaderId = leader.Id
Visibility = (OrganizationVisibility)request.Visibility

Reply:
Fixed in: 2dab9f8650b03dd242620092e163429fe17e90a0